app/src/main/java/com/android/shelfLife/ui/recipes/GenerateRecipe.kt

```diff
@@ -0,0 +1,166 @@
+package com.android.shelfLife.ui.recipes
+
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import android.widget.Toast
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Button
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.shelfLife.model.foodFacts.FoodFacts
+import com.android.shelfLife.model.foodFacts.Quantity
+import com.android.shelfLife.model.foodItem.FoodItem
+import com.android.shelfLife.model.recipe.RecipePrompt
+import com.android.shelfLife.model.recipe.RecipeType
+import com.android.shelfLife.ui.navigation.NavigationActions
+import com.android.shelfLife.viewmodel.recipe.RecipeGenerationViewModel
+
+@Composable
+fun GenerateRecipeScreen(
+    navigationActions: NavigationActions,
+    generationViewModel: RecipeGenerationViewModel
+) {
+  val context = LocalContext.current
+
+  // States for recipe name and food items
+  var recipeName by remember { mutableStateOf("") }
```

Comment:
This wont work if the screen state is changed (i.e. user changes from dark mode to light mode all the data in these variables will be erased). To make them work you need to either pass them to the ViewModel (since the ViewModel remembers it regardless of the screens state) or use rememberSaveable.

Reply:
thanks for pointing this out!